task/backend/bolt/bolt.go

```diff
@@ -434,6 +447,10 @@ func (s *Store) FindTaskMetaByID(ctx context.Context, id platform.ID) (*backend.
 		return nil, err
 	}
 
+	if stm.LatestCompleted < s.minLatestCompleted {
```

Comment:
Looking at it now, my concern with this approach is that we're returning something different from what's in the task store.

I think it will be okay for now, so I'm approving. But we'll need to keep in mind that this comes at a risk of difficulty debugging later if what we're exposing to the higher layers doesn't match exactly what's stored on disk.

If we run into debugging that, and if it's that painful, we can refactor to the other discussed approach of modifying every task meta record at startup.

Reply:
I agree, It could be difficult to debug. We will have to keep an eye on it.